### PR TITLE
Polish the UI

### PR DESCRIPTION
**Pull request by Kekun**
This modernizes the UI a bit by using some features from libhandy.

Changes:
- isolate each view and give them their own header bar
- use `HdyApplicationWindow` to have cool looking rounded corners and easy animations
- use a `HdyViewSwitcher` instead of a `GtkStackSwitcher`
- move the Download button in the page to have a fully top-to-bottom flow
- vertically center the Audio and Video pages
- clamp the Audio and Video pages horizontally
- animate the header bars and not only the content

Capture d’écran de 2020-08-12 22-23-00

Capture d’écran de 2020-08-12 22-32-03

If you like the changes, I have a few more ideas after that to further polish the UI.

### CODE REVIEW

File: com.github.unrud.VideoDownloader.json [hunk 1]
```diff
@@ -112,6 +112,23 @@
...
+            "config-opts": [
+                "-Dexamples=false",
+                "-Dglade_catalog=disabled",
+                "-Dtests=false",
+                "-Dvapi=false"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.gnome.org/GNOME/libhandy.git"
```
Comment (Unrud) [inline com.github.unrud.VideoDownloader.json:128]: This should be a release tarball.

File: src/window.py [hunk 2]
```diff
@@ -183,27 +182,23 @@ def _on_activate_link(self, _, uri):
...
             elif mode == 'video':
+                self.video_download_wdg.grab_default()
                 self.video_url_wdg.grab_focus()
             else:
                 assert False
-        elif state in ['download', 'cancel']:
-            self.cancel_wdg.grab_focus()
-        elif state in ['success', 'error']:
-            self.back_wdg.grab_focus()
+        elif state == 'download':
```
Comment (Unrud) [inline src/window.py:197]: This should be
```python3
state in ['download', 'cancel']
```

**Comment by Unrud** [comment]
Unfortunately I was unable to get it to work with Snap.
Switching the Gnome extension to 3.34 causes the program to segfault.
**libhandy** doesn't compile because of some incompatibly with **libpango** (the Gnome extension ships version 1.44.6).

File: src/window.py (hunk 2 above)
Comment (Kekun) [reply src/window.py:197]: I didn't understand what the cancel state is, as there is no child named like that in the UI file and so the state variable should never hold this value. Is that child added somewhere and I didn't see it? Or maybe is the download page's name changed somehow?

File: com.github.unrud.VideoDownloader.json (hunk 1 above)
Comment (Kekun) [reply com.github.unrud.VideoDownloader.json:128]: Oh sure.

**Comment by Kekun** [comment]
> Unfortunately I was unable to get it to work with Snap.
> Switching the Gnome extension to 3.34 causes video-downloader to segfault.
> **libhandy** doesn't compile because of some incompatibly with **libpango** (the Gnome extension ships version 1.44.6).

It's weird because libhandy doesn't directly depend on Pango (only via other libs like GTK)…

I never worked with Snap, but libhandy could be included as a Meson subproject if it can help somehow.

File: src/window.py (hunk 2 above)
Comment (Unrud) [reply src/window.py:197]: The cancel state uses the download UI child:

https://github.com/Unrud/video-downloader/blob/eb12092c31bf1051d297363c7404669c081bb21e/src/window.py#L67-L69

**Comment by Unrud** [comment]
> It's weird because libhandy doesn't directly depend on Pango (only via other libs like GTK)…

It fails during generation of GIR.

**Comment by Unrud** [comment]
Merged in 9f2d3d2808b1217fc1dd92ddb2f15fb528a14e44

**Pull request closed without merging**